api/src/game_state_server.py

```diff
@@ -40,10 +73,18 @@ def __init__(self, room_id: str, initial_connection: Optional[Hashable] = None):
         self.game_state: Dict[str, Union[Ping, Token]] = {}
         self.id_to_positions: Dict[str, List[Tuple[int, int, int]]] = {}
         self.positions_to_ids: Dict[Tuple[int, int, int], str] = {}
+        self.tokens_by_icon_id: Dict[str, List[Token]] = {}
         self.clients = set()
         if initial_connection:
             self.clients.add(initial_connection)
 
+    def pop_token(self, token_id: str) -> Token:
+        token = self.game_state[token_id]
+        del self.game_state[token_id]
+        if type(token) != Token:
```

Comment:
If you use isinstance instead of type() then mypy will be able to tell you did a type check here

Reply:
I was curing mypy (and you) for not knowing that it was the right type after checking. Thanks for the tip.

Reply:
👍